### PR TITLE
Check against sr project

### PR DESCRIPTION
**Pull request by imobach**
IMHO when we check for bugzilla, fate... IDs, we should do it against `sr_project` diff file. In fact, we're checking the `sr_project` spec file to check if version was changed. Why not do it in the same way for the `changes` file?

Scenario:

* You submit a new version of a package but you forgot, for example, to add a file (it happens).
* Quickly you create a new PR to fix the issue (before the SR is accepted).
* The package will be rejected as no changes were included.

### CODE REVIEW

**Comment by mvidner** [comment]
Sorry I don't understand what problem is this solving. Is there a an example of a failed submission?

File: lib/tasks/check_changelog.rake [hunk 1]
```diff
@@ -28,7 +28,7 @@ namespace "check" do
       if version_changed?( "#{osc_checkout_dir}/#{package_name}.spec" )
         Dir.chdir(osc_checkout_dir) do
           # Tags described in https://github.com/openSUSE/osc-plugin-factory/blob/e12bc02e9817277335ce6adaa8e8d334d03fcc5d/check_tags_in_requests.py#L63
-          cmd = "osc -A '#{obs_api}' diff *.changes"
+          cmd = "osc -A '#{obs_api}' cat '#{obs_sr_project}' '#{package_name}' '#{package_name}.changes' | diff - '#{package_name}.changes'"
```
Comment (mvidner) [inline lib/tasks/check_changelog.rake]: the line is too long

**Comment by imobach** [comment]
@mvidner The explanation was quite confusing. Just see https://ci.opensuse.org/view/Yast/job/yast-yast2-master/325/console

Thanks!

File: package/rubygem-packaging_rake_tasks.changes [hunk 2]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Thu Mar 17 12:33:14 UTC 2016 - igonzalezsosa@suse.com
+
+- Compare with sr_project when checking *.changes file for
```
Comment (mvidner) [inline package/rubygem-packaging_rake_tasks.changes]: > When checking for bug IDs in *.changes, diff against obs_sr_project, not obs_project, to be consistent with the version checker.

**Comment by mvidner** [comment]
Thanks, now it makes sense. I've proposed a new changes message, the code looks good.

**Comment by imobach** [comment]
I've updated the message (it's pretty clear now, thanks).

**Comment by lslezak** [comment]
LGTM

**Comment by mvidner** [comment]
LGTM

**Comment by imobach** [comment]
Thanks both :)

**Comment by jreidinger** [comment]
@imobach previous one have also problem with versioning as this gem use semantic version and @schubi add feature, not just bug fix. So can you use it and increase version to 1.2.0? thanks

**Pull request merged**